lib/Basic/FileSystem.cpp

```diff
@@ -228,10 +228,10 @@ swift::areFilesDifferent(const llvm::Twine &source,
   if (destRegionErr)
     return convertDestinationError(destRegionErr);
 
-  if (0 == memcmp(sourceRegion.const_data(), destRegion.const_data(), size))
-    return FileDifference::SameContents;
+  if (memcmp(sourceRegion.const_data(), destRegion.const_data(), size)))
```

Comment:
Nitpick: Can you explicitly compare this to `!= 0`? Every time I use or read a use of `memcmp` I forget that it returns 0 if the regions are equal, and it helps to keep the explicit comparison to 0

Reply:
There's also an extra `)` here.